src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SimulatorEnclaveProvider.NetCoreApp.cs

```diff
@@ -84,9 +84,11 @@ internal override void CreateEnclaveSession(byte[] attestationInfo, ECDiffieHell
                         Buffer.BlockCopy(attestationInfo, attestationInfoOffset, trustedModuleDHPublicKeySignature, 0,
                             checked((int)sizeOfTrustedModuleDHPublicKeySignatureBuffer));
 
-                        ECParameters ecParams = KeyConverter.ECCPublicKeyBlobToParams(trustedModuleDHPublicKey);
-                        ECDiffieHellman enclaveDHKey = ECDiffieHellman.Create(ecParams);
-                        byte[] sharedSecret = clientDHKey.DeriveKeyFromHash(enclaveDHKey.PublicKey, HashAlgorithmName.SHA256);
+                        byte[] sharedSecret;
+                        using (ECDiffieHellman ecdh = KeyConverter.CreateECDiffieHellmanFromPublicKeyBlob(trustedModuleDHPublicKey))
```

Comment:
NIT: I think using statement could be simplified after [C#8](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8#using-declarations). Is there any specific reason we have used the older style?

Reply:
Done.